src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs

```diff
@@ -0,0 +1,19 @@
+using System.Net;
+using OrchardCore.DisplayManagement.Views;
+
+namespace OrchardCore.Diagnostics.ViewModels;
+
+public class HttpStatusCodeShapeViewModel : ShapeViewModel
+{
+    public HttpStatusCode HttpStatusCode { get; }
+
+    public HttpStatusCodeShapeViewModel(HttpStatusCode statusCode)
+    {
+        HttpStatusCode = statusCode;
+        Metadata.Type = "HttpStatusCode";
+
+        // The type name is 'HttpStatusCode', so any Http status code will be handled by the 'HttpStatusCode.cshtml' or 'HttpStatusCode.liquid'.
+        // However, assign an alternative for every 'StatusCode' to enable the customization for each Status (ex. 'NotFound.cshtml' or 'NotFound.liquid').
+        Metadata.Alternates.Add(statusCode.ToString());
```

Comment:
Maybe add the int value too as an alternate.

Reply:
https://andrewlock.net/netescapades-enumgenerators-a-source-generator-for-enum-performance/

Reply:
I added the 404 to the alternate. We'll look at the library and see how we can use it widely but in a separate PR.